cpp/include/rmm/device_scalar.hpp

```diff
@@ -161,7 +201,16 @@ class device_scalar {
    */
   [[nodiscard]] value_type value(cuda_stream_view stream) const
   {
-    return _storage.front_element(stream);
+    if (_host_bounce_buffer) {
+      // Case: Copying with pinned host memory
+      RMM_CUDA_TRY(cudaMemcpyAsync(
+        *_host_bounce_buffer, data(), sizeof(T), cudaMemcpyDefault, stream.value()));
```

Comment:
I prefer making the std::optional more obvious with _house_bounce_buffer.value()

Reply:
No, I mean instead of the `*` dereference operator, use `.value()`. And in the case where you have `**`, use `*buffer.value()`.